### PR TITLE
[YUNIKORN-2272] refactor TestSpecificUserAndGroupResourceUsage

### PR DESCRIPTION
**Pull request by brandboat**
### What is this PR for?
add missing asserting for existing user/group and rename methods in TestSpecificUserAndGroupResourceUsage

### What type of PR is it?
* [x] - Improvement

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2272

### How should this be tested?
covered by unit test

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: pkg/webservice/handlers_test.go [hunk 1]
```diff
@@ -1453,6 +1467,23 @@ func TestSpecificUserAndGroupResourceUsage(t *testing.T) {
 	assert.NilError(t, err, "Get User Resource Usage Handler request failed")
 	resp = &MockResponseWriter{}
 	getUserResourceUsage(resp, req)
+	assertUserExists(t, resp,
+		&dao.UserResourceUsageDAOInfo{
+			UserName: "testuser",
```
Comment (chia7712) [inline pkg/webservice/handlers_test.go:1472]: `groupsLimitsConfig` does not define user `testuser`. why it can get passed?

**Review by chia7712 — COMMENTED** [review]
@brandboat thanks for patch. overall LGTM

File: pkg/webservice/handlers_test.go (hunk 1 above)
Comment (brandboat) [reply pkg/webservice/handlers_test.go:1472]: It looks like this is intentional. manager.go#getUserTracker returns the requested user tracker and creates one if it does not exist.
https://github.com/apache/yunikorn-core/blob/3b5a65d085ff49319e4ce6705e91a416efea1d70/pkg/scheduler/ugm/manager.go#L556-L570

Comment (chia7712) [reply pkg/webservice/handlers_test.go:1472]: Ok, thanks

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**